# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/WebGLFramebuffer.h
# Commit: 7fea32057b92
# Full Hash: 7fea32057b920eac49960541efa2f3fea9074c19
# Author: Lee Salzman <lsalzman@mozilla.com>
# Date: 2022-02-12 09:47:43
# Regressor Bug: 1754130
# File Overlap Count: 2
# Description:
#   Bug 1754130 - Support presenting a WebGLFramebuffer to its own swap chain without opaque FB. r=aosmond,jgilbert
#   
#   Most of the support for presenting a WebGLFramebuffer to a swap chain existed as part of the
#   mechanism for opaque WebXR framebuffer support. However, such "opaque" framebuffer are meant
#   to be opaque in the sense that their attachments can't be inspected or changed, which does
# ==============================================================================

diff -r 1ac8f10e0025 -r 7fea32057b92 dom/canvas/WebGLFramebuffer.h
--- a/dom/canvas/WebGLFramebuffer.h	Fri Feb 11 15:36:30 2022 +0000
+++ b/dom/canvas/WebGLFramebuffer.h	Fri Feb 11 15:36:30 2022 +0000
@@ -144,8 +144,10 @@
   const GLuint mGLName;
   bool mHasBeenBound = false;
   const UniquePtr<gl::MozFramebuffer> mOpaque;
-  gl::SwapChain mOpaqueSwapChain;
   bool mInOpaqueRAF = false;
+  // Swap chain that may be used to present this framebuffer, for opaque
+  // framebuffers or other use cases. (e.g. DrawTargetWebgl)
+  gl::SwapChain mSwapChain;
 
  private:
   mutable uint64_t mNumFBStatusInvals = 0;